src/data/polynomial/field_division.lean

```diff
@@ -55,6 +55,18 @@ have h₁ : (leading_coeff q)⁻¹ ≠ 0 :=
   inv_ne_zero (mt leading_coeff_eq_zero.1 h),
 by rw [degree_mul, degree_C h₁, add_zero]
 
+theorem irreducible_of_monic {F : Type u} [field F] {p : polynomial F} (hp1 : p.monic)
+  (hp2 : p ≠ 1) (hp3 : ∀ f g : polynomial F, f.monic → g.monic → f * g = p → f = 1 ∨ g = 1) :
+  irreducible p :=
```

Comment:
`hp3` could be turned into part of an `iff` together with `irreducible p`, right?

Reply:
done